x-pack/plugins/rule_registry/server/rule_data_plugin_service/rule_data_plugin_service.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger } from 'kibana/server';
+
+import { RuleDataClient } from '../rule_data_client';
+import { IndexNames } from './index_names';
+import { IndexOptions } from './index_options';
+import { ResourceNames } from './resource_names';
+import { ResourceInstaller, Resources } from './resource_installer';
+
+interface ConstructorOptions {
+  getClusterClient: () => Promise<ElasticsearchClient>;
+  logger: Logger;
+  isWriteEnabled: boolean;
+  index: string;
+}
+
+export class RuleDataPluginService {
+  private readonly resourceNames: ResourceNames;
+  private readonly resourceInstaller: ResourceInstaller;
+  private readonly installCommonResources: () => Promise<void>;
+
+  constructor(private readonly options: ConstructorOptions) {
+    this.resourceNames = new ResourceNames({ indexPrefixFromConfig: options.index });
+    this.resourceInstaller = new ResourceInstaller({
+      resourceNames: this.resourceNames,
+      getClusterClient: options.getClusterClient,
+      logger: options.logger,
+      isWriteEnabled: options.isWriteEnabled,
+    });
+
+    this.installCommonResources = this.resourceInstaller.memoizeInstallation(Resources.common, () =>
+      this.resourceInstaller.installResourcesSharedBetweenAllIndices()
+    );
+  }
+
+  public getResourcePrefix() {
```

Comment:
Correct, it's not. I expect it to be useful for RBAC, because RBAC depends on https://github.com/elastic/kibana/issues/106432 (feature -> index names mapping must respect the index prefix).

@yctercero @dhurley14

Reply:
For visibility, I pinged Yara and she said it would be useful. Leaving this function in the code.